hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/associations/AssociationsTranslator.java

```diff
@@ -59,8 +58,9 @@ public AssociationsTranslator(@NonNull final AssociationsDecoder decoder) {
      */
     @Override
     public boolean matches(@NonNull final HtsCallAttempt attempt) {
-        return (attempt.isTokenRedirect() && matchesHrcSelector(attempt.selector()))
-                || (!attempt.isTokenRedirect() && matchesClassicSelector(attempt.selector()));
+        return (attempt.isTokenRedirect() && attempt.isSelector(HRC_ASSOCIATE, HRC_DISSOCIATE))
```

Comment:
This particular conditional expression is very awkward.  I must have missed commenting on it when it was originally written (previous lines 62-63).  Parenthesis or not I just don't like conditional expressions that have _both_ kinds of short-circuit logical operators (`||` and `&&`).  

Can this be rewritten?  Maybe a switch expression on `attempt.isTokenRedirect()`?  Maybe just use an `if` statement on `attempt.isTokenRedirect()`?

Reply:
Reworked the conditions